helper/logging/logger.go

```diff
@@ -112,6 +112,8 @@ func ParseLogFormat(format string) (LogFormat, error) {
 	}
 }
 
+// ParseLogLevel returns the hclog.Level that corresponds with the provided level string.
+// This differs hclog.LevelFromString in that it supports additional level strings.
```

Comment:
I think I took this code and refactored it into a function when adding the log file feature, so this is my bad and thanks for adding comments!

However, I think that the func should probably *not* include `notice` or `warning` as that hasn't (to my knowledge) been documented on our site. 

There's also differing info depending on whether you look at the docs for the [`server` command](https://developer.hashicorp.com/vault/docs/commands/server#log-level) or [Vault configuration](https://developer.hashicorp.com/vault/docs/configuration#log_level). 

`Trace, Debug, Error, Warn, Info.` vs `trace, debug, info, warn, *err*`.

```suggestion
// ParseLogLevel differs hclog.LevelFromString in that it supports fewer log level strings.
```

[My PR](https://github.com/hashicorp/vault/pull/18031) I have open at the moment to add rotation to log files has updates to the docs to make them all standardised, and I will also update `ParseLogLevel` to not handle `notice`, `warning`  or `""` which defaults to `info`.